azure-mgmt-datalake-store/src/test/java/com/microsoft/azure/management/datalake/store/DataLakeStoreAccountOperationsTests.java

```diff
@@ -7,57 +7,87 @@
 package com.microsoft.azure.management.datalake.store;
 
 import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.datalake.store.models.CapabilityInformation;
 import com.microsoft.azure.management.datalake.store.models.DataLakeStoreAccount;
 import com.microsoft.azure.management.datalake.store.models.DataLakeStoreAccountBasic;
 import com.microsoft.azure.management.datalake.store.models.DataLakeStoreAccountUpdateParameters;
+import com.microsoft.azure.management.datalake.store.models.NameAvailabilityInformation;
+import com.microsoft.azure.management.datalake.store.models.OperationListResult;
 import org.junit.Assert;
 import org.junit.Test;
 
 import java.util.HashMap;
-import java.util.List;
 
-public class DataLakeStoreAccountOperationsTests extends DataLakeStoreManagementTest {
+public class DataLakeStoreAccountOperationsTests extends DataLakeStoreManagementTest
+{
     @Test
-    public void canCreateGetUpdateDeleteAdlsAccount() throws Exception {
+    public void canCreateGetUpdateDeleteAdlsAccount() throws Exception
+    {
         String adlsAcct = generateRandomResourceName("adlsacct", 15);
 
+        // Ensure that the account name is available
+        NameAvailabilityInformation checkNameGetResponse = dataLakeStoreAccountManagementClient.accounts().checkNameAvailability(
+                environmentLocation.name(),
+                adlsAcct
+        );
+
+        Assert.assertTrue(checkNameGetResponse.nameAvailable());
+
         // Create
         DataLakeStoreAccount createParams = new DataLakeStoreAccount();
         createParams.withLocation(environmentLocation.name());
         createParams.withTags(new HashMap<String, String>());
         createParams.getTags().put("testkey", "testvalue");
```

Comment:
This API is not following the fluent pattern; fluent pattern would look like datalakeStoreAccounts.defineAccount(...).withLocation(...).withTag(...).create(). Is it a test for auto-generated layer and fluent layer is not there yet?